src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ReferenceCountedDisposable.cs

```diff
@@ -223,13 +222,12 @@ public ValueTask DisposeAsync()
         /// Represents a weak reference to a <see cref="ReferenceCountedDisposable{T}"/> which is capable of
         /// obtaining a new counted reference up until the point when the object is no longer accessible.
         /// </summary>
+        /// <remarks>
+        /// This value type holds a single field, which is not subject to torn reads/writes.
+        /// </remarks>
         public struct WeakReference
```

Comment:
readonly struct?

Reply:
Do we actually need to wrap BoxedReferenceCount now? Could BoxedReferenceCount  be renamed to WeakReference?

Reply:
➡️ Now marked readonly

Reply:
Oh, never mind, I guess we don't want to expose the object since it's used for locking